consensus/src/persistent_liveness_storage.rs

```diff
@@ -250,12 +397,25 @@ impl RecoveryData {
                 )
             })?;
 
+        // If execution pool is enabled, use the window_root, else use the commit_root
```

Comment:
this seems very un-intuitive, from the comment on window_root_block, it's only None when it's genesis, not when execution pool is disabled?